crates/codegen/runtime/cargo/crate/src/runtime/parser/parser_support/parser_function.rs

```diff
@@ -85,14 +85,9 @@ where
                     TerminalKind::UNRECOGNIZED
                 };
                 let node = Node::terminal(kind, input.to_string());
-                let tree = if no_match.kind.is_none() || start.utf8 == 0 {
-                    node
-                } else {
-                    trivia_nodes.push(Edge::anonymous(node));
-                    Node::nonterminal(no_match.kind.unwrap(), trivia_nodes)
-                };
+                trivia_nodes.push(Edge::anonymous(node));
                 ParseOutput {
-                    parse_tree: tree,
+                    parse_tree: Rc::new(NonterminalNode::new(no_match.kind.unwrap(), trivia_nodes)),
```

Comment:
if we are always unwrapping `no_match.kind`, should it still be an `Option<>`?

Reply:
`NoMatch` is being used in the `default()` function with `None`

Reply:
Here is an example of how it's used today:

https://github.com/NomicFoundation/slang/blob/f257eae7828a09222436c0201c7a8a48106c300f/crates/codegen/runtime/generator/src/parser/codegen/precedence_parser_definition.rs#L45

Reply:
IIUC, in the case you mentioned, the kind is already known ([`nonterminal_name`](https://github.com/NomicFoundation/slang/blob/f257eae7828a09222436c0201c7a8a48106c300f/crates/codegen/runtime/generator/src/parser/codegen/precedence_parser_definition.rs#L34C13-L34C29)). Right? Can't we just initialize it with that expected kind, to prevent another code path from accidentally leaving a `None` here?